coreneuron/io/phase2.cpp

```diff
@@ -0,0 +1,1216 @@
+    // it may happen that Point_process structures will be made unnecessary
+    // by factoring into NetCon.
+
+#if CHKPNTDEBUG
+    ntc.pnttype = new int[nnetcon];
+    ntc.pntindex = new int[nnetcon];
+    memcpy(ntc.pnttype, pnttype, nnetcon * sizeof(int));
+    memcpy(ntc.pntindex, pntindex, nnetcon * sizeof(int));
+#endif
+    for (int i = 0; i < nnetcon; ++i) {
+        int type = pnttype[i];
+        if (type > 0) {
+            int index = pnt_offset[type] + pntindex[i];  /// Potentially uninitialized pnt_offset[],
+                                                         /// check for previous assignments
+            NetCon& nc = nt.netcons[i];
+            nc.target_ = nt.pntprocs + index;
+            nc.active_ = true;
+        }
+    }
+
+    int extracon_target_nweight = 0;
+    if (nrn_setup_extracon > 0) {
+        // Fill in the extracon target_ and active_.
+        // Simplistic.
+        // Rotate through the pntindex and use only pnttype for ProbAMPANMDA_EMS
+        // (which happens to have a weight vector length of 5.)
+        // Edge case: if there is no such synapse, let the target_ be nullptr
+        //   and the netcon be inactive.
+        // Same pattern as algorithm for extracon netcon_srcgid above in phase1.
+        int extracon_target_type = nrn_get_mechtype("ProbAMPANMDA_EMS");
+        assert(extracon_target_type > 0);
+        extracon_target_nweight = corenrn.get_pnt_receive_size()[extracon_target_type];
+        int j = 0;
+        for (int i = 0; i < nrn_setup_extracon; ++i) {
+            bool active = false;
+            for (int k = 0; k < nnetcon; ++k) {
+                if (pnttype[j] == extracon_target_type) {
+                    active = true;
+                    break;
+                }
+                j = (j + 1) % nnetcon;
+            }
+            NetCon& nc = nt.netcons[i + nnetcon];
+            nc.active_ = active ? 1 : 0;
+            if (active) {
+                nc.target_ = nt.pntprocs + (pnt_offset[extracon_target_type] + pntindex[j]);
+            } else {
+                nc.target_ = nullptr;
+            }
+        }
+    }
+
+    {
+        nt.n_weight = weights.size();
+        int nweight = nt.n_weight;
+        // weights in netcons order in groups defined by Point_process target type.
+        nt.n_weight += nrn_setup_extracon * extracon_target_nweight;
+        nt.weights = (double*)ecalloc_align(nt.n_weight, sizeof(double));
+        std::copy(weights.begin(), weights.end(), nt.weights);
+
+        int iw = 0;
+        for (int i = 0; i < nnetcon; ++i) {
+            NetCon& nc = nt.netcons[i];
+            nc.u.weight_index_ = iw;
+            if (pnttype[i] != 0) {
+                iw += corenrn.get_pnt_receive_size()[pnttype[i]];
+            } else {
+                iw += 1;
+            }
+        }
+        assert(iw == nweight);
+
+#if CHKPNTDEBUG
+        ntc.delay = new double[nnetcon];
+        memcpy(ntc.delay, delay, nnetcon * sizeof(double));
```

Comment:
nitpick : replace with std::copy for concistency?

Reply:
we must add a test enabling this `CHKPNTDEBUG`

Reply:
Done